feat: implement form instance management with Jotai in Form component

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Added lifecycle management for dialog and form instances, enabling more robust tracking of open dialogs and forms.
	- Improved navigation logic to prevent leaving pages with open dialogs or unsaved form changes, including confirmation prompts when necessary.
	- Introduced a new translation key for close confirmation descriptions.
- **Bug Fixes**
	- Enhanced modal and dialog handling to reduce the risk of data loss and ensure smoother user experience during navigation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: implement form instance management with Jotai in Form component

- Added formInstancesAtom to manage form instances using Jotai.
- Integrated form instance tracking in FormWrapper with useEffect for lifecycle management.
- Enhanced Bootstrap component to utilize form instances for improved dialog handling.
- Updated translations to include confirmation messages for unsaved changes across multiple languages.

limichange committed:
feat: integrate dialog instance management with Jotai

- Added dialog instance tracking using Jotai in the Dialog component.
- Implemented addDialogInstance and removeDialogInstance functions for managing dialog instances.
- Enhanced Bootstrap component to utilize dialog instances for improved dialog handling.
- Exported dialog instance management functions for broader accessibility.

limichange committed:
Merge branch 'x' into fix-dialog-form-2

sidmorizon commented on packages/components/src/composite/Dialog/index.tsx:
```diff
@@ -56,6 +56,7 @@ import {
   DialogTitle,
   SetDialogHeader,
 } from './Header';
+import { addDialogInstance, removeDialogInstance } from './jotai';
```
为啥要用状态容器，这里没有涉及到 UI 层响应式更新，而是在函数里需要访问，全局变量就行了。 

状态容器会引起 UI 刷新

limichange replied:
好的。

limichange committed:
Merge branch 'x' into fix-dialog-form-2

limichange committed:
refactor: replace Jotai-based instance management with local state in Dialog and Form components

- Introduced dialogInstances and formInstances management using local arrays instead of Jotai atoms.
- Updated Dialog and Form components to utilize new addDialogInstance and removeDialogInstance functions for instance tracking.
- Removed Jotai-related code and exports from both Dialog and Form components for cleaner implementation.

limichange committed:
Merge branch 'x' into fix-dialog-form-2

limichange committed:
refactor: update Dialog component to use local instance management

- Replaced Jotai exports with local instance management functions for dialog instances.
- Cleaned up imports and exports in the Dialog component for improved clarity and organization.

limichange committed:
Merge branch 'x' into fix-dialog-form-2

limichange committed:
Merge branch 'x' into fix-dialog-form-2
